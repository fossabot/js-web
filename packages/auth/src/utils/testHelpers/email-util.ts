export function getRandomEmail(domain = 'yopmail.com'): string {
  const chars = 'abcdefghijklmnopqrstuvwxyz1234567890';
  let email = '';
  // eslint-disable-next-line no-plusplus
  for (let i = 0; i < 15; i++) {
    email += chars[Math.floor(Math.random() * chars.length)];
  }
  return `${email}@${domain}`;
}
