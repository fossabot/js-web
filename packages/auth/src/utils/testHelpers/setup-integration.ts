import { Test, TestingModule } from '@nestjs/testing';
import { UsersService } from '@seaccentral/core/dist/user/users.service';
import { SeedService } from '@seaccentral/core/dist/seed/seed.service';
import { Connection } from 'typeorm';
import { AppModule } from '../../app.module';

export async function setupApp() {
  const app = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();

  await app.get(Connection).synchronize(true);
  await app.get(UsersService).populateRoles();
  await app.get(SeedService).createPasswordSetting();
  await app.get(SeedService).createMockedPlans();

  return app;
}

export async function teardownApp(app: TestingModule) {
  await app.get(Connection).close();
  await app.close();
}
