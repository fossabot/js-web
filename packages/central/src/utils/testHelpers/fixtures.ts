import { INestApplication } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { JwtService } from '@nestjs/jwt';
import { UsersService } from '@seaccentral/core/dist/user/users.service';
import { User } from '@seaccentral/core/dist/user/User.entity';
import { EntityManager, Repository } from 'typeorm';
import { addHours } from 'date-fns';
import { Role } from '@seaccentral/core/dist/user/Role.entity';
import { UserRole } from '@seaccentral/core/dist/user/UserRole.entity';
import { Policy } from '@seaccentral/core/dist/user/Policy.entity';
import { GOD_MODE } from '@seaccentral/core/dist/access-control/constants';
import { RolePolicy } from '@seaccentral/core/dist/user/RolePolicy.entity';
import { AdminService } from '../../admin/admin.service';

export const createSession = async (app: INestApplication) => {
  const adminService = app.get(AdminService);
  const userService = app.get(UsersService);
  const jwtService = app.get(JwtService);
  const configService = app.get(ConfigService);

  await userService.populateRoles();
  await adminService.createLoginSetting();
  const user = await userService.createRandomUser();
  const adminRole = await app
    .get(EntityManager)
    .getRepository(Role)
    .findOneOrFail({ name: 'Admin' });
  const allAccessPolicy = await app
    .get(EntityManager)
    .getRepository(Policy)
    .create({ name: GOD_MODE.GRANT_ALL_ACCESS })
    .save();
  await app
    .get(EntityManager)
    .getRepository(RolePolicy)
    .insert({ role: adminRole, policy: allAccessPolicy });
  await app
    .get(EntityManager)
    .getRepository(UserRole)
    .insert({ user, role: adminRole });

  const payload = { userId: user.id };
  const accessToken = jwtService.sign(payload, {
    secret: configService.get('JWT_SECRET'),
    expiresIn: `${configService.get('JWT_EXPIRATION_TIME_IN_SECONDS')}s`,
  });

  return { user, accessToken };
};

export const createNewUser = async (app: INestApplication) => {
  const userService = app.get(UsersService);
  return userService.createRandomUser();
};

export const lockUser = async (app: INestApplication, id: string) => {
  const userRepository: Repository<User> = app.get('UserRepository');
  const user = await userRepository.findOne(id);
  if (!user) {
    return null;
  }
  user.accessFailedCount = 10;
  user.isLockedOut = true;
  user.lockoutEndDateUTC = addHours(Date.now(), 5);
  return user.save();
};
