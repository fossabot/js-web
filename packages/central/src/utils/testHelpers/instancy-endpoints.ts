import { rest } from 'msw';
import { format, URL } from 'url';

function endpoints(basePath: string) {
  return [
    rest.post(
      `${basePath}/GetUserInfoByEmailForAllSites`,
      async (req, res, ctx) => {
        return res(ctx.xml(''));
      },
    ),
    rest.post(`${basePath}/CreateUser`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/UpdateUser`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/CopyUsertoASite`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/UserMembershipDetails`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/RenewMemebrship`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/UpdateMemebrship`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/UpdateMemebrship`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
    rest.post(`${basePath}/MembershipDetails`, async (req, res, ctx) => {
      return res(ctx.xml(''));
    }),
  ];
}

export const apiC23 = endpoints(
  format(new URL(process.env.INSTANCY_C23_BASE_URL as string), {
    search: false,
  }),
);
export const apiPkgAllAccess = endpoints(
  format(new URL(process.env.INSTANCY_ALL_ACCESS_PACKAGE_BASE_URL as string), {
    search: false,
  }),
);
export const apiPkgOnline = endpoints(
  format(new URL(process.env.INSTANCY_ONLINE_PACKAGE_BASE_URL as string), {
    search: false,
  }),
);
export const apiPkgVirtual = endpoints(
  format(new URL(process.env.INSTANCY_VIRTUAL_PACKAGE_BASE_URL as string), {
    search: false,
  }),
);
