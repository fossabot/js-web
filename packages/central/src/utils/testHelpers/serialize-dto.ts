/**
 * For simulate going through nestjs validation pipe for dto.
 * But by using dto instance as parameter for convinent type-checking.
 * This will transform and validate according to class-validator/class-transformer decorator applied to dto.
 */

import { classToClass } from 'class-transformer';
import { validate } from 'class-validator';

export async function serializeDto<T extends Record<string, any>>(dto: T) {
  const transformedDto = classToClass(dto);
  const validationError = await validate(transformedDto);
  expect(validationError).toEqual([]);

  return transformedDto;
}
