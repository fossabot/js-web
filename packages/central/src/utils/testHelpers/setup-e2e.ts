import { INestApplication } from '@nestjs/common';
import { Test } from '@nestjs/testing';
import { Connection } from 'typeorm';
import { SeedService } from '@seaccentral/core/dist/seed/seed.service';
import { UsersService } from '@seaccentral/core/dist/user/users.service';
import { AppModule } from '../../app.module';

export async function clearDbModels(app: INestApplication) {
  await app.get(Connection).synchronize(true);
}

export async function setupServer() {
  const moduleFixture = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();

  const app = moduleFixture.createNestApplication();
  await app.init();

  return app;
}

export async function beforeEachServer(app: INestApplication) {
  await clearDbModels(app);
  await app.get(SeedService).createPasswordSetting();
  await app.get(UsersService).populateRoles();
}

export async function teardownServer(app: INestApplication) {
  await app.get(Connection).close();
  await app.close();
}
