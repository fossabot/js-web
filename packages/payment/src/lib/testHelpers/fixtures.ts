import { INestApplication } from '@nestjs/common';
import { ConfigService } from '@nestjs/config';
import { JwtService } from '@nestjs/jwt';
import { UsersService } from '@seaccentral/core/dist/user/users.service';

export const createSession = async (app: INestApplication) => {
  const userService = app.get(UsersService);
  const configService = app.get(ConfigService);
  const jwtService = app.get(JwtService);

  await userService.populateRoles();
  const user = await userService.createRandomUser();

  const payload = { userId: user.id };
  const accessToken = jwtService.sign(payload, {
    secret: configService.get('JWT_SECRET'),
    expiresIn: '1d',
  });

  return { user, accessToken };
};
