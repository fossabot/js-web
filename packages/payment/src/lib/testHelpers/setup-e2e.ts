import { INestApplication } from '@nestjs/common';
import { Test } from '@nestjs/testing';
import { Connection } from 'typeorm';
import { SeedService } from '@seaccentral/core/dist/seed/seed.service';
import { AppModule } from '../../app.module';

export async function clearDbModels(app: INestApplication) {
  await app.get(Connection).synchronize(true);
}

export async function setupServer() {
  const moduleFixture = await Test.createTestingModule({
    imports: [AppModule],
  }).compile();

  const app = moduleFixture.createNestApplication();
  await clearDbModels(app);
  await app.get(SeedService).createPasswordSetting();
  await app.init();

  return app;
}

export async function teardownServer(app: INestApplication) {
  await app.get(Connection).close();
  await app.close();
}
