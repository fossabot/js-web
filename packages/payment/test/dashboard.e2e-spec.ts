import { HttpStatus, INestApplication } from '@nestjs/common';
import request from 'supertest';
import { getUnixTime, addDays } from 'date-fns';
import { setupServer, teardownServer } from '../src/lib/testHelpers/setup-e2e';
import { createSession } from '../src/lib/testHelpers/fixtures';

describe('DashboardController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    app = await setupServer();
  });

  afterEach(async () => {
    await teardownServer(app);
  });

  it('as an admin, I should be able to filter by date range for payment data', async () => {
    const { accessToken } = await createSession(app);
    const now = new Date();
    const query = {
      fromDate: getUnixTime(addDays(now, -7)),
      toDate: getUnixTime(addDays(now, 7)),
    };
    const { body } = await request(app.getHttpServer())
      .get('/v1/dashboard/payment')
      .set('auth_token', accessToken)
      .query(query)
      .expect(HttpStatus.OK);

    const { data } = body;
    expect(data.newUser).toBeGreaterThanOrEqual(0);
    expect(data.renewUser).toBeGreaterThanOrEqual(0);
    expect(data.expiredUser).toBeGreaterThanOrEqual(0);
    expect(data.unsuccessfulPayment).toBeGreaterThanOrEqual(0);
    expect(Array.isArray(data.packageType)).toBe(true);
    expect(Array.isArray(data.memberType)).toBe(true);
    expect(data.revenue).toBeGreaterThanOrEqual(0);
  });

  it('as an admin, I should be able to export a csv file from payment data', async () => {
    const { accessToken } = await createSession(app);
    const now = new Date();
    const query = {
      fromDate: getUnixTime(addDays(now, -7)),
      toDate: getUnixTime(addDays(now, 7)),
    };

    const response = await request(app.getHttpServer())
      .get('/v1/dashboard/payment/csv')
      .query(query)
      .set('auth_token', accessToken)
      .expect(HttpStatus.OK)
      .buffer()
      .parse((responseStream, callback) => {
        const chunks: Buffer[] = [];
        responseStream.on('data', (chunk) => chunks.push(Buffer.from(chunk)));
        responseStream.on('error', (err) => callback(err, null));
        responseStream.on('end', () =>
          callback(null, Buffer.concat(chunks).toString('utf8')),
        );
      });
    expect(response.headers['content-disposition']).toContain('.csv');
    expect(response.headers['content-type']).toContain('csv');
    expect(response.body).toMatchSnapshot();
  });
});
