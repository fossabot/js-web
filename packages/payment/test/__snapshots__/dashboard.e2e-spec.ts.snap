// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardController (e2e) as an admin, I should be able to export a csv file from payment data 1`] = `
"newUser,renewUser,expiredUser,unsuccessfulPayment,packageType,memberType,revenue
0,0,0,0,,,0"
`;
